src/System.Memory/tests/ReadOnlySpan/CompareTo.cs

```diff
@@ -0,0 +1,279 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Globalization;
+using System.Threading;
+using Xunit;
```

Comment:
in general for tests, I prefer to edit the tests in [https://github.com/dotnet/corefx/blob/master/src/System.Globalization/tests/CompareInfo](https://github.com/dotnet/corefx/blob/master/src/System.Globalization/tests/CompareInfo) and inject the Span cases inside the test cases. by doing that we'll ensure the string and the span behavior is identical and in the future any added more test data will get automatically applied to both the strings and the spans

Reply:
There is some overlap here: https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/StringTests.cs#L589

Also, the CompareInfo tests are testing CompareOptions, which Span APIs do not have.

Reply:
Would be better then add the span tests to the string tests you have pointed at. I just want to ensure the string and span behaving exactly same.

Reply:
That is where I borrowed the tests from. I had to make some minor adjustments which is why it doesn't seem worth the effort to move the tests back.

Also, span has specific tests that are unique to it, so this way all the span tests live in one place (and one assembly).

Reply:
I didn't mean to borrow the test but inject the span tests into the string tests. it'll pay off when adding more test cases or data to these tests. Maintaining 2 places which we need to guarantee the consistency would be hard.

For Unique Span tests, that fine to split these.

Reply:
If the string APIs start calling the span APIs instead, we would get the coverage without having to update the tests.

I can update the tests if it turns out we can't call the span APIs from the string APIs due to perf or other reason.

Reply:
See https://github.com/dotnet/corefx/issues/28238